pkg/controller/replication/replication_controller.go

```diff
@@ -456,6 +474,37 @@ func (rm *ReplicationManager) worker() {
 	}
 }
 
+func (rm *ReplicationManager) removeControllerRef(pods []*api.Pod, rcUID types.UID) []error {
+	var errlist []error
+	for _, pod := range pods {
+		for _, owner := range pod.OwnerReferences {
+			glog.V(2).Infof("patching pod %s_%s to remove its controllerRef to %s/%s:%s",
+				pod.Namespace, pod.Name, owner.APIVersion, owner.Kind, owner.Name)
+			deleteOwnerRefPatch := fmt.Sprintf(`{"metadata":{"ownerReferences":[{"$patch":"delete","uid":"%s"}],"uid":"%s"}}`, owner.UID, pod.UID)
+			err := rm.podControl.PatchPod(pod.Namespace, pod.Name, []byte(deleteOwnerRefPatch))
+			if err != nil {
+				if errors.IsNotFound(err) {
+					// If the pod no longer exists, ignore it.
+					continue
+				} else if errors.IsInvalid(err) {
+					// Invalid error will be returned in two cases: 1. the pod
+					// has no owner reference, 2. the uid of the pod doesn't
+					// match, which means the pod is deleted and then recreated.
+					// In both cases, the error can be ignored.
+
+					// TODO: If the pod has owner references, but none of them
+					// has the owner.UID, server will silently ignore the patch.
+					// Investigate why.
```

Comment:
It didn't cause any problem to the RC manager or the GC. I was trying to say the behavior of the server is different depending on if the OwnerReferences list is empty, upon receiving a deletion patch, server returns 422 if the list is empty, but returns 200 if the list is not empty, but doesn't contain the target ownerReference.